core/include/vecmem/memory/resources/terminal/malloc_resource.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "vecmem/memory/resources/memory_resource.hpp"
+
+#include <memory>
+#include <cstdlib>
```

Comment:
I believe you should only use `<cstddef>` here. Just to ensure that `std::size_t` is defined. Even though `vecmem::memory_resource` would ensure even that... So you may as well just not add any std includes here.